src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -210,13 +211,16 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     Dictionary<KeyValueWatcher, KeyValueChange> keyValueChanges = null;
                     List<KeyValueChange> changedKeyValuesCollection = null;
                     bool refreshAll = false;
+                    StringBuilder logInfoBuilder = new StringBuilder();
+                    StringBuilder logDebugBuilder = new StringBuilder();
 
                     await ExecuteWithFailOverPolicyAsync(availableClients, async (client) =>
                         {
                             applicationSettings = null;
                             keyValueChanges = new Dictionary<KeyValueWatcher, KeyValueChange>();
                             changedKeyValuesCollection = null;
                             refreshAll = false;
+                            Uri endpoint = _configClientManager.GetEndpointForClient(client);

```

Comment:
Can you clear both string builders here? If failover occurred, we dont want log statements from failed endpoints to be present in log builders.